src/sentry/roles/manager.py

```diff
@@ -1,25 +1,45 @@
+from __future__ import annotations
+
+import abc
+import re
 from collections import OrderedDict
-from typing import Dict, Iterable, Mapping, Optional, Tuple
+from dataclasses import dataclass
+from typing import Any, Dict, FrozenSet, Generic, Iterable, Mapping, Sequence, Tuple, Type, TypeVar
 
+from sentry.utils import warnings
 
-class Role:
-    def __init__(
-        self,
+
+def _normalize_whitespace(text: str) -> str:
+    return re.sub(r"\s+", " ", text.strip())
+
+
+R = TypeVar("R", bound="Role")
```

Comment:
non blocking - R is a little obfuscated 🤷 
Not super clear this is a Role type down below

Reply:
Single letters are a convention for generic type parameters (for examples, see https://docs.python.org/3/library/typing.html and https://mypy.readthedocs.io/en/stable/generics.html).